dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/ITextCompletion.cs

```diff
@@ -21,4 +22,15 @@ public interface ITextCompletion
         string text,
         CompleteRequestSettings requestSettings,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Creates a completion for the prompt and settings.
+    /// </summary>
+    /// <param name="text">The prompt to complete.</param>
+    /// <param name="requestSettings">Request settings for the completion API</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Streaming content of the text generated by the remote model</returns>
+    public IAsyncEnumerable<string> CompleteStreamAsync(string text,
```

Comment:
what about using a collection of streams?

Reply:
Not sure what you mean by collection of streams, is that a parallel request, or literally multiple results at once?

Reply:
Followed up offline. We have discussed a strategy to further evolve the interface to support returning more responses and more metadata, without breaking all callers. I agree with getting this in first so that users can start experimenting with streaming soon, and we'll continue to invest in a richer public API.